public/services/requests/sql.ts

```diff
@@ -12,10 +12,18 @@ export class SQLService {
     this.http = http;
   }
 
-  fetch = async (params: DirectQueryRequest, errorHandler?: (error: any) => void) => {
+  fetch = async (
+    params: DirectQueryRequest,
+    dataSourceMDSId?: string,
+    errorHandler?: (error: any) => void
+  ) => {
+    const query = {
+      dataSourceMDSId,
+    };
     return this.http
       .post('/api/observability/query/jobs', {
         body: JSON.stringify(params),
+        query,
```

Comment:
We should merge the `dataSourceMDSId` in post request `body`, similar to https://github.com/opensearch-project/dashboards-observability/blob/3dbe5c8e8ee0b5659bd59e265e9d4c1994b723c0/server/routes/custom_panels/panels_router.ts#L338-L344 and update this router accordingly